### PR TITLE
Handle project creation with invalid label scenario

### PR DESCRIPTION
**Pull request by chamilaadhi**
Handle project creation with invalid label scenario. When a project is created with an invalid label, show valid error message on the CLI and delete the partially created project. Same goes to project creation with api name version combination.

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java [hunk 1]
```diff
@@ -268,7 +269,21 @@ public void execute() {
             apis = service.getAPIs(label, accessToken);
         } else {
             ExtendedAPI api = service.getAPI(apiName, version, accessToken);
-            apis.add(api);
+            if(api != null) {
```
Comment (mushthaq33) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java]: Shall we format this?

Comment (chamilaadhi) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java]: done

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java [hunk 2]
```diff
@@ -819,4 +819,40 @@ private static void saveScript(String content, File path) throws IOException {
...
+        } catch (IOException e) {
+	    //not throwing the error because deleting faild project is not
+            // a critical task. This can be deleted manually if not used.
+            logger.error("Failed to delete project : {} ", projectPath, e);
+        }
+
+    }
+
+    private static void delete(File file) throws IOException {
+
```
Comment (ruks) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java]: unnecessary spaces right?

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java [hunk 3]
```diff
@@ -268,7 +269,21 @@ public void execute() {
...
             ExtendedAPI api = service.getAPI(apiName, version, accessToken);
-            apis.add(api);
+            if (api != null) {
+                apis.add(api);
+            }
+        }
+        if (apis == null || (apis != null && apis.isEmpty())) {
+            // Delete folder
+            GatewayCmdUtils.deleteProject(workspace + File.separator + projectName);
+            String errorMsg = "";
```
Comment (ruks) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java]: Initialization is redundant

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java (hunk 2 above)
Comment (chamilaadhi) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/GatewayCmdUtils.java]: removed

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java (hunk 3 above)
Comment (chamilaadhi) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java]: removed

**Pull request merged**